### DISCUSSION TITLE
Ratelimit in Fastapi

### DISCUSSION THREAD

**Discussion by parsa10** [fastapi/fastapi#8689] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/", dependencies=[Depends(RateLimiter(times=2, seconds=5))])
async def index():
    return {"msg": "Hello World"}
```

### Description

There is no official way to add ratelimit in path operation functions , it would be good if we could limit requests to a specific rate , for example: a client can only request to "/" path 3 times in 60 seconds.
NOTE: Im aware of libraries made for handling this , but they are not trustable and useful enough.

### Wanted Solution

i would like a dependency class to handle ratelimits , Both for **_HTTP_** And **_WEBSOCKET_** Requests

### Wanted Code

```python
@app.get("/", dependencies=[Depends(RateLimiter(times=2, seconds=5))])
async def index():
    return {"msg": "Hello World"}
```

### Operating System

Linux

### FastAPI Version

0.70.0

### Python Version

3.9.0

**Comment by mgurg** [comment]
I recently saw two projects solving this problem:
- [https://github.com/laurentS/slowapi](https://github.com/laurentS/slowapi)
- [https://github.com/long2ice/fastapi-limiter](https://github.com/long2ice/fastapi-limiter)

**Comment by parsa10** [comment]
> I recently saw two projects solving this problem:
> 
> * https://github.com/laurentS/slowapi
> * https://github.com/long2ice/fastapi-limiter

**NOTE: Im aware of libraries made for handling this , but they are not trustable and useful enough.**

which one did you use and which one was better ? 
None of them support websocket

**Comment by dstlny** [comment, marked as answer]
> > I recently saw two projects solving this problem:
> > 
> > * https://github.com/laurentS/slowapi
> > * https://github.com/long2ice/fastapi-limiter
> 
> **NOTE: Im aware of libraries made for handling this , but they are not trustable and useful enough.**
> 
> which one did you use and which one was better ? None of them support websocket

What isn't trustable about them?
They're both literally open-source. You can go ahead and read the code line-by-line if you wanted to to make sure they aren't doing anything nefarious. Which I doubt they are.

**Comment by michaelsizonenko** [comment]
> > I recently saw two projects solving this problem:
> > 
> > * https://github.com/laurentS/slowapi
> > * https://github.com/long2ice/fastapi-limiter
> 
> **NOTE: Im aware of libraries made for handling this , but they are not trustable and useful enough.**
> 
> which one did you use and which one was better ? None of them support websocket

You managed to find a solution which support websockets

**Comment by WMRamadan** [comment]
I think this features should be built into FastAPI.

**Comment by aybruhm** [comment]
> > I recently saw two projects solving this problem:
> > 
> > * https://github.com/laurentS/slowapi
> > * https://github.com/long2ice/fastapi-limiter
> 
> **NOTE: Im aware of libraries made for handling this , but they are not trustable and useful enough.**
> 
> which one did you use and which one was better ? 
> None of them support websocket

Dude, did you go through the documentation of [fastapi-limiter](https://github.com/long2ice/fastapi-limiter)? 'cause it does support websocket. 

![Screenshot_20221225-063208~2.png](https://user-images.githubusercontent.com/55067204/209457982-04ed7e4f-0b96-4f71-b4ef-43f4d6aa7825.png)
